app/assets/javascripts/helpers/govspeak-fetch.js

```diff
@@ -0,0 +1,20 @@
+window.GovspeakFetch = { }
+
+window.GovspeakFetch.getBody = function (text) {
+  var path = document.location.pathname
+  var segments = path.split('/')
+  var documentSegment = segments[1]
```

Comment:
This looks better - we can be more confident about the URL structure. Still hoping we can be more explicit...how about: `document.location.pathname.match("/documents/([^/]*)/")[1]` to get the edition ID?

We should also throw some kind of textual exception so it's easier for devs to debug this function if it somehow gets called from a non-document URL.

Reply:
Hm.. it's not really clear to me how the regex helps our case, as after we extract the edition ID, we'll have to prefix it back with the `/documents/` string anyway, then append the `govspeak-preview`, but gave it a go if you think it makes more sense.

Reply:
Agreed the difference is quite subtle, but I can see two reasons why a regex approach would be preferable, unless there are cross-browser issues with it:

   - We want to document the kind of URL we're expecting, so the supported paths are clear by reading the code. At the moment, we're just fishing out the 1st and 2nd segments, which could be anything. Another way to document this would be with a comment.
   - We want to make our code fail in a helpful way. Since we don't check the first segment is '/documents', it would be hard to debug if this code is run with an unexpected `window.location`. Another way would be to check `segments[1] == '/documents'`.

I suggested a regex approach because it should be a more concise way to make the approach robust and clear. Because we're intentionally coupling/hard-coding our JS, we should also make sure it gives a helpful error if that coupling breaks in future.

Happy to talk about this more - I was hoping to catch you on Friday and go through it together.

Reply:
No cross-browser concerns. I get the reasoning better now, thanks for explaining it. Added a commit as you suggested and I've kicked the CI again as it was failing because of Publishing API.